### ISSUE TITLE
[BUG] Opensearch stack is stuck in DELETE_FAILED status

### ISSUE THREAD

**Issue by manojrajpurohit** [awslabs/autonomous-driving-data-framework#82]
**Describe the bug**
When we try to deploy and then remove modules/integration/ddb-to-opensearch/ module , it puts the opensearch stack addf-ros-image-demo-core-opensearch in DELETE_FAILED status. Thus, we not able to deploy modules/integration/ddb-to-opensearch/  again now. The DELETE_FAILED status is because AWS::EC2::SecurityGroup resource of stack is stuck in DELETE_FAILED, refer below message `resource _sg-id_has a dependent object (Service: AmazonEC2; Status Code: 400; Error Code: DependencyViolation; Request ID: _masked_; Proxy: null)`

**To Reproduce**

1. Deploy ADDF with OpenSearch , refer manifest here https://github.com/manojrajpurohit/autonomous-driving-data-framework/blob/trigger-spark-on-eks/manifests/ros-image-demo/integration-modules.yaml
2. Remove manifests/ros-image-demo/optional-modules.yaml from ADDF manifest and deploy again, this will attempt to delete opensearch and put the stack in DELETE_FAILED state
3. Include opensearch module again in ADDF and deploy, Opensearch will not be deployed and remains in DELETE_FAILED state

**Expected behavior**
Opensearch stack should be deployed and deleted without entering DELTE_FAILED state

**Screenshots**
![os failed](https://user-images.githubusercontent.com/33446857/206187797-2e30589d-7586-4f3c-8b80-29d98f8024be.JPG)

**Additional context**
I think this is because the security group is associated in Network interfaces, post which CloudFormation is unable to delete the security group because of the dependency. refer link https://aws.amazon.com/premiumsupport/knowledge-center/ec2-find-security-group-resources/

**Comment by dgraeber** [comment]
@manojrajpurohit 
It appears that the Security Group tied to the Opensearch Deployment has a reference to some entity that has not been removed from it yet(?).  Can you please go into the account, find the referred to SG and see what resources are using it by trying to manually deleting the SG.  Once you are able to delete the SG, then delete the CFN stack and rerun `seedfarmer apply`

**Comment by dgraeber** [comment]
@manojrajpurohit Also - check if the stack has the Opensearch proxy has been deployed and not destroyed.  I would like to know how the customer got here (i.e. they had a deployment and wanted to swap out OpenSearch, they were trying to delete the entire deployment, etc).

**Comment by manojrajpurohit** [comment]
Indeed the security group was used in Network Interface. I guess these Network Interface were created as part of modules/demo-only/opensearch-proxy/
So I did the following 

1. commented manifests/ros-image-demo/integration-modules.yaml in deployment.yaml and executed seedfarmer, thus, modules/demo-only/opensearch-proxy/ and modules/integration/ddb-to-opensearch/ stacks were deleted
2. deleted opensearch stack from CloudFormation console. **Was able to delete it this time.**
3. Commented opensearch in manifest and rerun seedfarmer - deployment failed (I thought this was required to put seedfarmer in right state since I had deleted opensearch from console)
4. Included opensearch , modules/demo-only/opensearch-proxy/ and modules/integration/ddb-to-opensearch/  in manifest and rerun seedfarmer - all 3 stacks are successfully created 

so I was somehow able to fix this issue. 

However, I think, the point to remember is after deploying opensearch, opensearch proxy and ddb-to-opensearch, we should not try to remove opensearch alone , rather all 3 should be removed. I am not sure if this understanding is correct. But if this is correct then I am wondering whether there is a way to hard specify dependency check between stacks.

Anyways closing closing this bug now.
